manager/assets/modext/workspace/lexicon/lexicon.grid.js

```diff
@@ -248,30 +247,46 @@ Ext.extend(MODx.grid.Lexicon,MODx.grid.Grid,{
     	this.loadWindow(btn, e, o);
     }
     ,reloadFromBase: function() {
-    	Ext.Ajax.timeout = 0;
-    	var topic = '/workspace/lexicon/reload/';
-        this.console = MODx.load({
-           xtype: 'modx-console'
-           ,register: 'mgr'
-           ,topic: topic
-        });
+        var tb = this.getTopToolbar(),
+        namespace = tb.getComponent('namespace').getValue(),
+        topicLexicon = tb.getComponent('topic').getValue(),
+        language = tb.getComponent('language').getValue(),
+        topic = '/workspace/lexicon/reload/';
 
-        this.console.on('complete',function(){
-            this.refresh();
-        },this);
-        this.console.show(Ext.getBody());
+        MODx.msg.confirm({
+            text: _('lexicon_revert_confirm', {
+                namespace: namespace
+                ,topic: topicLexicon
+                ,language: language
+            })
+            ,url: this.config.url
+            ,params: {
+                action: 'Workspace/Lexicon/ReloadFromBase'
+                ,register: 'mgr'
+                ,topic: topic
+                ,namespace: namespace
+                ,topicLexicon: topicLexicon
+                ,language: language
+            }
+            ,listeners: {
+                'success': {
+                    fn:function() {
+                        this.console = MODx.load({
+                            xtype: 'modx-console'
+                            ,register: 'mgr'
+                            ,topic: topic
```

Comment:
The following suggestion is much less than it looks like, mostly I'd like to encourage that as contributors make changes we apply better code style and updated syntax/grammar along the way; in this case:
- Move commas from beginning to end of statements
- Remove unneeded quotes from object keys (this will be most of the time, unless the key contains a character such as a dash and others that then require the key to be quoted)

I also suggest minor naming tweaks to the topic variables in an attempt to clarify them a little. The clash between the completely unrelated `topic` variables for Lexicons and the Registry threw me off at first.

```suggestion
                namespace: namespace,
                topic: topic,
                language: language
            }),
            url: this.config.url,
            params: {
                action: 'Workspace/Lexicon/ReloadFromBase',
                register: 'mgr',
                topic: registryTopic,
                namespace: namespace,
                lexiconTopic: topic,
                language: language
            },
            listeners: {
                success: {
                    fn: function() {
                        this.console = MODx.load({
                            xtype: 'modx-console',
                            register: 'mgr',
                            topic: registryTopic
```

Reply:
Changes like this make diffs on PRs very hard to read (tho github helps). It's a tricky problem and each approach has its own drawbacks, but I'm leaning towards doing one massive automated code style swoop at some point to give it all consistency.

Reply:
Wasn't there a discussion in the core-contributors channel where we sort of decided against this?

Reply:
Yes, such global changes should be made in another PR, let's leave it as it is for now.